packages/webapp/webapp_server.js

```diff
@@ -799,12 +803,22 @@ function runWebAppServer() {
       return getBoilerplateAsync(
         request,
         archKey
-      ).then(boilerplate => {
-        var statusCode = res.statusCode ? res.statusCode : 200;
+      ).then(({ start, stream, end, statusCode, headers: newHeaders }) => {
+        if (!statusCode) {
+          statusCode = res.statusCode ? res.statusCode : 200;
+        }
+        if (newHeaders) {
+          headers = {...headers, ...newHeaders };
+        }
+
         res.writeHead(statusCode, headers);
-        res.write(boilerplate);
-        res.end();
-      }, error => {
+        res.write(start);
+        stream.pipe(res, { end: false })
+        stream.on("end", () => {
+          res.write(end);
+          res.end();
+        })
```

Comment:
For example, if `start` and `end` were part of the `stream`, this code could just be
```js
stream.pipe(res, { end: true });
```